Source/WebGPU/WebGPU/ComputePipeline.mm

```diff
@@ -86,14 +86,37 @@ static MTLSize metalSize(auto workgroupSize)
     if (!function)
         return ComputePipeline::createInvalid(*this);
 
-    MTLComputePipelineReflection *reflection;
-    bool hasBindGroups = pipelineLayout.numberOfBindGroupLayouts() > 0;
-    auto computePipelineState = createComputePipelineState(m_device, function, pipelineLayout, computeInformation, label, &reflection, hasBindGroups ? MTLPipelineOptionNone : MTLPipelineOptionArgumentInfo);
+    if (!pipelineLayout.numberOfBindGroupLayouts() && entryPointInformation.defaultLayout) {
+        Vector<Vector<WGPUBindGroupLayoutEntry>> bindGroupEntries;
+        addPipelineLayouts(bindGroupEntries, entryPointInformation.defaultLayout);
```

Comment:
Why doesn't the function just return `bindGroupEntries`?

Why does this have to be a method on `Device`?

Reply:
It’s a member function of Device since it’s shared by createRenderPipeline and createComputePipeline

Reply:
We could return a vector but for RenderPipeline we would need to merge the fragment into the vertex vector. So instead of performing a merge I used an out parameter.